app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/vm.rb

```diff
@@ -104,6 +107,15 @@ def console_supported?(type)
     return true if type.upcase == 'VNC'
   end
 
+  def console_url
+    crn = CGI.escape(ext_management_system.pcloud_crn.values.join(":"))
```

Comment:
As annoying as this is, `CGI.escape` should not be used because it doesn't handle things correctly...prefer `ERB::Util.url_encode`.

Aside, I'm tempted to put this in more_core_extensions so we don't have to think about this as much in the future.

Reply:
I recently used `CGI.escape` [here](https://github.com/ManageIQ/manageiq-providers-ibm_cloud/blob/master/app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager.rb#L57). Sounds like it should be avoided in general?